core/controllers/reader.py

```diff
@@ -1027,3 +1028,43 @@ def get(self):
             'question_dicts': question_dicts
         })
         self.render_json(self.values)
+
+
+class LearnerAnswerDetailsSubmissionHandler(base.BaseHandler):
+    """Handles the learner answer details submission."""
+
+    @acl_decorators.can_play_entity
+    def put(self, entity_type, entity_id):
+        """"Handles the PUT requests. Stores the answer details submitted
+        by the learner.
+        """
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        interaction_id = self.payload.get('interaction_id')
+        if entity_type == feconf.ENTITY_TYPE_EXPLORATION:
+            state_name = self.payload.get('state_name')
+            state_reference = (
+                stats_services.get_state_reference_for_exploration(
+                    entity_id, state_name))
+            if interaction_id != exp_services.get_interaction_id_for_state(
+                    entity_id, state_name):
+                raise utils.InvalidInputException(
+                    'Interaction id given does not match with the '
+                    'interaction id of the state')
+        elif entity_type == feconf.ENTITY_TYPE_QUESTION:
+            state_reference = (
+                stats_services.get_state_reference_for_question(entity_id))
+            if interaction_id != (
+                    question_services.get_interaction_id_for_question(
+                        entity_id)):
+                raise utils.InvalidInputException(
+                    'Interaction id given does not match with the '
+                    'interaction id of the question')
+
```

Comment:
What if the entity type is neither exploration or question? should we throw an error?